frontends/concrete-python/examples/prime-match/prime-match.py

```diff
@@ -0,0 +1,275 @@
+
+    def __repr__(self):
+        return f"{repr(self.order_type)} {self.order_quantity} of {repr(self.order_symbol)}"
+
+    @staticmethod
+    def random(skiplist: Set[OrderSymbol]) -> "Order":
+        order_type = random.choice(list(OrderType))
+        order_symbol = random.choice(list(set(OrderSymbol) - skiplist))
+        order_quantity = random.randint(MINIMUM_ORDER_QUANTITY, MAXIMUM_ORDER_QUANTITY)
+        return Order(order_type, order_symbol, order_quantity)
+
+
+def prime_match(
+    bank_order_types,
+    bank_order_symbols,
+    bank_order_quantities,
+    client_order_types,
+    client_order_symbols,
+    client_order_quantities,
+):
+    with fhe.tag("comparing-order-types"):
+        types_differ = bank_order_types.reshape(-1, 1) != client_order_types
+
+    with fhe.tag("comparing-order-symbols"):
+        symbols_match = bank_order_symbols.reshape(-1, 1) == client_order_symbols
+
+    with fhe.tag("checking-if-order-can-be-filled"):
+        can_fill = symbols_match & types_differ
+
+    with fhe.tag("calculating-matching-order-quantity"):
+        matching_quantity = np.minimum(
+            bank_order_quantities.reshape(-1, 1), client_order_quantities
+        )
+
+    with fhe.tag("calculating-filled-order-quantity"):
+        filled_quantity = fhe.multivariate(lambda x, y: x * y)(can_fill, matching_quantity)
+
+    with fhe.tag("creating-result"):
+        result = fhe.zeros((NUMBER_OF_BANK_ORDERS + NUMBER_OF_CLIENT_ORDERS, 1 + 1 + 1))
+
+        result[0:NUMBER_OF_BANK_ORDERS, 0] = bank_order_types
+        result[0:NUMBER_OF_BANK_ORDERS, 1] = bank_order_symbols
+
+        result[NUMBER_OF_BANK_ORDERS:, 0] = client_order_types
+        result[NUMBER_OF_BANK_ORDERS:, 1] = client_order_symbols
+
+        result[:NUMBER_OF_BANK_ORDERS, 2] = np.sum(filled_quantity, axis=1)
+        result[NUMBER_OF_BANK_ORDERS:, 2] = np.sum(filled_quantity, axis=0)
+
+        return result
+
+
+inputset = [
+    (
+        np.random.randint(0, len(OrderType), size=(NUMBER_OF_BANK_ORDERS,)),
+        np.array(
+            [int(symbol) for symbol in random.sample(list(OrderSymbol), NUMBER_OF_BANK_ORDERS)]
+        ),
+        np.random.randint(
+            MINIMUM_ORDER_QUANTITY,
+            MAXIMUM_ORDER_QUANTITY + 1,
+            size=(NUMBER_OF_BANK_ORDERS,),
+        ),
+        np.random.randint(0, len(OrderType), size=(NUMBER_OF_CLIENT_ORDERS,)),
+        np.array(
+            [int(symbol) for symbol in random.sample(list(OrderSymbol), NUMBER_OF_CLIENT_ORDERS)]
+        ),
+        np.random.randint(
+            MINIMUM_ORDER_QUANTITY,
+            MAXIMUM_ORDER_QUANTITY + 1,
+            size=(NUMBER_OF_CLIENT_ORDERS,),
+        ),
+    )
+    for _ in range(1000)
+]
+configuration = fhe.Configuration(
+    enable_unsafe_features=True,
+    use_insecure_key_cache=True,
+    insecure_key_cache_location=".keys",
+    fhe_simulation=True,
```

Comment:
`fhe_simulation=True`: do we really execute in FHE? I would say Yes since timings are long but checking. CC @umut-sahin

Reply:
ok tx